src/ServiceManagement/Profile/Commands.Profile/Account/AddAzureAccount.cs

```diff
@@ -90,7 +121,64 @@ public override void ExecuteCmdlet()
                 }
 
                 WriteObject(account.ToPSAzureAccount());
-            } 
+            }
+        }
+
+        private IAzureAccount GetAccountFromSPN(AzureAccount account, IAzureEnvironment environment, SecureString password)
+        {
+            IAzureAccount tenantAccount = new AzureAccount();
+            CopyAccount(account, tenantAccount);
+            WriteDebugWithTimestamp("AzureSession.Instance.AuthenticationFactory.Authenticate");
+            var tenantToken = AzureSession.Instance.AuthenticationFactory.Authenticate(
```

Comment:
Try to write the full type name, if it's not clear on the right side of the assignment.